### PR TITLE
Item 6803: Query metadata editor migration

### PR DESCRIPTION
**Pull request by labkey-ankurj**
#### Related PRs:
* https://github.com/LabKey/platform/pull/885
* https://github.com/LabKey/labkey-ui-components/pull/178

### CODE REVIEW

File: src/labkey/Query.ts [hunk 1]
```diff
@@ -57,4 +60,36 @@ export {
...
+    schemaName?: string
+    failure?: (error?: any) => any
+    success?: (data?: any) => any
+}
+
+/**
+ * Gets QueryMetadata in the shape of a Domain design.
+ */
+export function getQueryMetadata(config: GetQueryMetadataOptions): void {
+    let options: GetQueryMetadataOptions = arguments.length > 1 ? {
```
Comment (labkey-nicka) [inline src/labkey/Query.ts]: I couldn't find a version of this in Query.js. Assuming this is a new endpoint we don't need to have "backwards compatibility" code in here to match arguments. The `config` itself will suffice. Let's remove the `arguments` parsing and rename `config` to `options`.

File: src/labkey/Query.ts [hunk 2]
```diff
@@ -57,4 +60,36 @@ export {
     URL_COLUMN_PREFIX,
     validateQuery,
     Visualization
+}
+
+export interface GetQueryMetadataOptions {
+    containerPath?: string
```
Comment (labkey-nicka) [inline src/labkey/Query.ts]: For the `getQueryMetadataOptions` interface let's make the following changes:

* At minimum `schemaName` and `queryName` are required properties.
* Documentation needs to be added for each property.
* Alphabetical declaration of interface properties.

File: src/labkey/Query.ts [hunk 3]
```diff
@@ -25,6 +25,9 @@ import { deleteRows, insertRows, saveRows, selectDistinctRows, selectRows, trunc
 import * as SQL from './query/experimental/SQL'
 import * as Visualization from './query/Visualization'
 import { Filter } from './filter/Filter'
+import {request} from "./Ajax";
```
Comment (labkey-nicka) [inline src/labkey/Query.ts]: I'd like to discuss moving the implementation out of `labkey/Query.ts` and into a subfile of `src/labkey/query`. It allows for `labkey/Query.ts` to act more as a exportable namespace free of implementation.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 ## 0.0.38 - 2020-02-27
 - Doc updates (migrated from legacy API)
 
+## 0.0.38-fb-queryMetadataEditor-0.0 - 2020-02-27
+- Item 6083: QueryMetadata Editor migration
```
Comment (labkey-nicka) [inline CHANGELOG.md]: nit: Consider rewording release comment to be more contextually relevant to labkey-api-js:

```
- Item 6083: Add Query.getQueryMetadata endpoint
```

**Comment by cnathe** [comment]
@labkey-ankurj @labkey-nicka This new query-getQueryEditorMetadata API is very likely only ever going to be used for the new QueryMetadataEditor.tsx component in the query module right? If so, is there a specific reason we want it here in labkey-api-js vs just putting the Ajax.request() directly in the query module so that it sits next to the QueryMetadataEditor.tsx component? That component has the query-saveQueyrMetadata Ajax.request anyway.

**Comment by labkey-nicka** [comment]
@cnathe @labkey-ankurj I'd definitely agree not putting it in labkey-api-js at all if we don't intend to expose it as a JS api (and wasn't explicitly planned by the story).

**Comment by labkey-ankurj** [comment]
I agree, I will be removing this and closing this branch.

**Pull request closed without merging**